src/Sources/SalesForce/SalesForceSource.cs

```diff
@@ -0,0 +1,264 @@
+                nameof(TimeoutException) => Directive.Restart,
+                nameof(HttpRequestException) => Directive.Restart,
+                _ => Directive.Stop
+            });
+
+            this.currentJob = Option<SalesForceJob>.None;
+            this.nextJobLocator = Option<string>.None;
+
+
+            this.SetHandler(source.Out, this.PullChanges, this.Finish);
+        }
+
+        private void Finish(Exception cause)
+        {
+            if (cause is not null && cause is not SubscriptionWithCancelException.NonFailureCancellation)
+            {
+                this.FailStage(cause);
+            }
+
+            this.httpClient.Dispose();
+        }
+
+        public override void PreStart()
+        {
+            this.UpdateSchema();
+        }
+
+        public void UpdateSchema()
+        {
+            var newSchema = this.source.jobProvider.GetSchema(httpClient, this.source.entityName).Result;
+
+            if (newSchema.IsEmpty)
+            {
+                this.Log.Warning("Could not update schema");
+            }
+            else
+            {
+                var schemaUnchanged =
+                    SalesForceEntity.SalesForceEntityComparer.Equals(this.entitySchema, newSchema.Value);
+                this.entitySchema = (this.entitySchema == null, schemaEquals: schemaUnchanged) switch
+                {
+                    (true, _) or (false, true) => newSchema.Value,
+                    (false, false) => throw new SchemaMismatchException()
+                };
+            }
+
+        }
+
+        private void CreateNewJob()
+        {
+            var job = this.source.jobProvider.CreateJob(httpClient, this.entitySchema).Result;
+
+            if (job.IsEmpty)
+            {
+                this.Log.Warning("Could not create job");
+            }
+            else
+            {
+                this.currentJob = job;
+            }
+            this.ScheduleOnce(TimerKey, TimeSpan.FromSeconds(1));
+        }
+
+        private void UpdateJobStatus()
+        {
+            var response = this.source.jobProvider.GetJobStatus(httpClient, this.currentJob.Value).Result;
+
+            if (response.IsEmpty)
+            {
+                this.Log.Warning("Could not create job");
+            }
+            else
+            {
+                this.currentJob = response;
+            }
+            this.ScheduleOnce(TimerKey, TimeSpan.FromSeconds(1));
+        }
+
+        private void ProcessResult()
+        {
+
+            var (rows, nextJobLocator) = this.source.jobProvider.GetJobResult(httpClient, this.currentJob.Value, this.entitySchema, this.nextJobLocator).Result;
+            this.nextJobLocator = nextJobLocator;
+            this.EmitMultiple(this.source.Out, rows);
+
+            if (nextJobLocator.IsEmpty)
+            {
+                this.currentJob = Option<SalesForceJob>.None;
+                this.ScheduleOnce(TimerKey, this.source.changeCaptureInterval);
```

Comment:
Something is not working here.
The stream keeps trying to fetch data instead of waiting 🤔

Reply:
The reason why this issue is happening is how Akka actually works.

Every Akka stream is a set of an objects (_stages_) running in parallel. Each stage can push an element to the downstream stage or pull an element from the upstream stage. When a sink is scheduled for execution and has no element inside its processing que, it will **pull** the source to obtain a new element for processing.

If we look in the execution stack when execution hits the `PullChanges` method, we can observe two kinds of stack:
- when the next element is pulled by the sink:
![image](https://github.com/SneaksAndData/arcane-framework/assets/5614941/3ca4e515-48a3-46a8-b7f7-d60ef73974eb)

- when the pull was caused by timer:
![image](https://github.com/SneaksAndData/arcane-framework/assets/5614941/e0dc1eeb-8006-47f0-85e2-8a004d134a4b)

To ensure this source operates correctly, we can examine the [source code](https://github.com/akkadotnet/akka.net/blob/851a72b361431d62dfa56f51ee8a11f99d896aab/src/core/Akka.Streams/Dsl/Pulse.cs#L51)  of the `Pulse` stage. This stage's function is essentially very similar to what you need to accomplish: it attempts to retrieve data from a certain data source, and if data is available, it pushes it to the subsequent stage. However, if data is not available or if the sink pulls from your source too frequently, the appropriate response is to do nothing.